R/qc_cvs.R

```diff
@@ -0,0 +1,174 @@
+
+      return(plot)
+    }
+    if(plot_style == "density")
+    {
+      plot <- ggplot2::ggplot(result) +
+        ggplot2::geom_density(aes(x = .data$values, col = .data$type), size = 1) +
+        ggplot2::theme_bw() +
+        ggplot2::ylab("Density") +
+        ggplot2::labs(title = "Coefficients of variation") +
+        ggplot2::xlab("Coefficient of variation [%]") +
+        ggplot2::scale_x_continuous(limits = c(0, 200)) +
+        ggplot2::scale_color_manual(values = c("#8D8D8D",
+                                              "#5680C1",
+                                              "#B96DAD",
+                                              "#64CACA",
+                                              "#81ABE9",
+                                              "#F6B8D1",
+                                              "#99F1E4",
+                                              "#9AD1FF",
+                                              "#548BDF",
+                                              "#A55098",
+                                              "#3EB6B6",
+                                              "#87AEE8",
+                                              "#CA91C1",
+                                              "#A4E0E0",
+                                              "#1D4F9A",
+                                              "#D7ACD2",
+                                              "#49C1C1"))
+
+      return(plot)
+    }
+      if(plot_style == "violin")
+      {
+        plot <- ggplot2::ggplot(result) +
+          ggplot2::geom_violin(aes(x = .data$type, y = .data$values, fill = .data$type)) +
+          ggplot2::theme_bw() +
+          ggplot2::ylab("Coefficient of variation [%]") +
+          ggplot2::labs(title = "Coefficients of variation") +
+          ggplot2::theme(axis.text.x = element_text(angle = 75, hjust = 1)) +
+          ggplot2::scale_y_continuous(limits = c(0, 200)) +
+          ggplot2::scale_fill_manual(values = c("#8D8D8D",
+                                                "#5680C1",
+                                                "#B96DAD",
+                                                "#64CACA",
+                                                "#81ABE9",
+                                                "#F6B8D1",
+                                                "#99F1E4",
+                                                "#9AD1FF",
+                                                "#548BDF",
+                                                "#A55098",
+                                                "#3EB6B6",
+                                                "#87AEE8",
+                                                "#CA91C1",
+                                                "#A4E0E0",
+                                                "#1D4F9A",
+                                                "#D7ACD2",
+                                                "#49C1C1"))
+        return(plot)
```

Comment:
```suggestion
        plot <- ggplot2::ggplot(result, aes(x = .data$type, y = .data$values, fill = .data$type)) +
          ggplot2::geom_violin(na.rm = TRUE) +
          ggplot2::geom_boxplot(width = 0.15, fill = "white", na.rm = TRUE) +
          ggplot2::theme_bw() +
          ggplot2::ylab("Coefficient of variation [%]") +
          ggplot2::labs(title = "Coefficients of variation") +
          ggplot2::theme(axis.text.x = element_text(angle = 75, hjust = 1)) +
          ggplot2::scale_y_continuous(limits = c(0, 200)) +
          ggplot2::scale_fill_manual(values = c("#8D8D8D",
                                                "#5680C1",
                                                "#B96DAD",
                                                "#64CACA",
                                                "#81ABE9",
                                                "#F6B8D1",
                                                "#99F1E4",
                                                "#9AD1FF",
                                                "#548BDF",
                                                "#A55098",
                                                "#3EB6B6",
                                                "#87AEE8",
                                                "#CA91C1",
                                                "#A4E0E0",
                                                "#1D4F9A",
                                                "#D7ACD2",
                                                "#49C1C1"))
        return(plot)
```
I would just use it like this. and you can add this na.rm also to the other plots, that will suppress the warning message. Then it will be only produced once.